raiden/network/transport/matrix/client.py

```diff
@@ -156,7 +156,8 @@ def _send(self, method, path, *args, **kwargs):
                 )
                 gevent.sleep(delay)
         else:
-            raise last_ex
+            if last_ex:
```

Comment:
@ulope Not sure here, but it seems like this can be `None` and that would throw a `TypeError`.

Reply:
I don’t see how it could. The else block of a for loop is only executed it the loop exhausts it’s iterator (e.g. no break or return in this case) is called inside the loop.

Reply:
I'm not sure I'm getting the point.

`retry_delay = lambda: repeat(1)` will never exhaust, right? So why would we need the `else`  branch anyways?

Reply:
The lambda is only the fallback default, but you're still right. Even the full implementation furns this effectively into an infinite loop. 

I guess since we don't necessarily control the caller one could argue that it's good defensive practice to catch the else case here even if it probably will never occur.